### ISSUE TITLE
New issue - Exception: Additions to this Web site have been blocked.

### ISSUE THREAD

**Issue by julesdadrummer** [Barbarur/NovaPoint#19]
Not sure if this is a NovaPoint issue or something changed on the SharePoint side of things. Let me know. Thanks!

2024-06-02 11:48:53 - [NPLogger] - Solution properties
2024-06-02 11:48:53 - [NPLogger] - ========== ========== ==========
2024-06-02 11:48:53 - [NPLogger] - OnlyUserList: False
2024-06-02 11:48:53 - [NPLogger] - AllUsers: False
2024-06-02 11:48:53 - [NPLogger] - IncludeUserUPN: lperez@redacted.qc.ca
2024-06-02 11:48:53 - [NPLogger] - IncludeExternalUsers: False
2024-06-02 11:48:53 - [NPLogger] - IncludeEveryone: False
2024-06-02 11:48:53 - [NPLogger] - IncludeEveryoneExceptExternal: False
2024-06-02 11:48:53 - [NPLogger] - RemoveAdmin: False
2024-06-02 11:48:53 - [NPLogger] - AllSiteCollections: True
2024-06-02 11:48:53 - [NPLogger] - IncludePersonalSite: False
2024-06-02 11:48:53 - [NPLogger] - IncludeShareSite: True
2024-06-02 11:48:53 - [NPLogger] - OnlyGroupIdDefined: False
2024-06-02 11:48:53 - [NPLogger] - SiteUrl: 
2024-06-02 11:48:53 - [NPLogger] - IncludeSubsites: False
2024-06-02 11:48:53 - [NPLogger] - ListOfSitesPath: 
2024-06-02 11:48:53 - [NPLogger] - IncludeAdmins: False
2024-06-02 11:48:53 - [NPLogger] - IncludeSiteAccess: True
2024-06-02 11:48:53 - [NPLogger] - IncludeUniquePermissions: True
2024-06-02 11:48:53 - [NPLogger] - AllLists: True
2024-06-02 11:48:53 - [NPLogger] - IncludeLists: True
2024-06-02 11:48:53 - [NPLogger] - IncludeLibraries: True
2024-06-02 11:48:53 - [NPLogger] - IncludeHiddenLists: False
2024-06-02 11:48:53 - [NPLogger] - IncludeSystemLists: False
2024-06-02 11:48:53 - [NPLogger] - ListTitle: 
2024-06-02 11:48:53 - [NPLogger] - FolderRelativeUrl: 
2024-06-02 11:48:53 - [NPLogger] - ========== ========== ==========
2024-06-02 11:48:53 - [NPLogger] - Solution has started, please wait to the end
2024-06-02 11:48:53 - [AppInfo] - Adding cached token
2024-06-02 11:48:53 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024-06-02 11:48:53 - [AppInfo] - Access Token expiration time: 2024-06-02 1:10:16 PM +00:00
2024-06-02 11:48:53 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/sites/root
2024-06-02 11:48:53 - [GraphAPIHandler] - Successful response
2024-06-02 11:48:53 - [Appinfo] - Hostname: redacted.sharepoint.com
2024-06-02 11:48:53 - [Appinfo] - Domain: redacted
2024-06-02 11:48:53 - [AppInfo] - SPO Admin URL 'https://redacted-admin.sharepoint.com'
2024-06-02 11:48:53 - [AppInfo] - SPO -my URL 'https://redacted-my.sharepoint.com'
2024-06-02 11:48:53 - [AppInfo] - SPO root URL 'https://redacted.sharepoint.com'
2024-06-02 11:48:53 - [GetAADUser] - Getting Signed-in user
2024-06-02 11:48:53 - [AppInfo] - Got access token from memory
2024-06-02 11:48:53 - [AppInfo] - Access Token expiration time: 2024-06-02 1:10:16 PM +00:00
2024-06-02 11:48:53 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/me
2024-06-02 11:48:54 - [GraphAPIHandler] - Successful response
2024-06-02 11:48:54 - [SPOSiteCollectionCSOM] - Getting Site Collections; IncludePersonalSite 'False', Group ID Defined 'False'
2024-06-02 11:48:54 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:54 - [AppInfo] - Adding cached token
2024-06-02 11:48:54 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024-06-02 11:48:54 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:57 - [SPOSiteCollectionCSOM] - Collected 1422 Site Collections...
2024-06-02 11:48:57 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:57 - [AppInfo] - Got access token from memory
2024-06-02 11:48:57 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:58 - [SPOSiteCollectionCSOM] - Collected 1467 Site Collections...
2024-06-02 11:48:58 - [SPOSiteCollectionCSOM] - Collected 1440 Site Collections
2024-06-02 11:48:58 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://redacted.sharepoint.com/sites/m-4286'
2024-06-02 11:48:58 - [SPOSiteCollectionAdminCSOM] - Processing 'https://redacted.sharepoint.com/sites/m-4286' setting 'admin-sp@redacted.qc.ca' IsSiteAdmin 'True'
2024-06-02 11:48:58 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:58 - [AppInfo] - Got access token from memory
2024-06-02 11:48:58 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:58 - [SPOSiteUserCSOM] - Getting user with email 'lperez@redacted.qc.ca' from Site 'https://redacted.sharepoint.com/sites/m-4286'
2024-06-02 11:48:58 - [AppInfo] - Getting Access Token for root site https://redacted.sharepoint.com
2024-06-02 11:48:58 - [AppInfo] - Adding cached token
2024-06-02 11:48:58 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024-06-02 11:48:58 - [AppInfo] - Access Token expiration time: 2024-06-02 12:50:03 PM +00:00
2024-06-02 11:48:58 - [SPOSiteUserCSOM] - User with email 'lperez@redacted.qc.ca' no found in Site 'https://redacted.sharepoint.com/sites/m-4286'
2024-06-02 11:48:58 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://redacted.sharepoint.com/sites/m-4114'
2024-06-02 11:48:58 - [SPOSiteCollectionAdminCSOM] - Processing 'https://redacted.sharepoint.com/sites/m-4114' setting 'admin-sp@redacted.qc.ca' IsSiteAdmin 'True'
2024-06-02 11:48:58 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:58 - [AppInfo] - Got access token from memory
2024-06-02 11:48:58 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:58 - [SPOSiteUserCSOM] - Getting user with email 'lperez@redacted.qc.ca' from Site 'https://redacted.sharepoint.com/sites/m-4114'
2024-06-02 11:48:58 - [AppInfo] - Getting Access Token for root site https://redacted.sharepoint.com
2024-06-02 11:48:58 - [AppInfo] - Got access token from memory
2024-06-02 11:48:58 - [AppInfo] - Access Token expiration time: 2024-06-02 12:50:03 PM +00:00
2024-06-02 11:48:59 - [SPOSiteUserCSOM] - User with email 'lperez@redacted.qc.ca' no found in Site 'https://redacted.sharepoint.com/sites/m-4114'
2024-06-02 11:48:59 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://redacted.sharepoint.com/sites/PrsentationsGBM'
2024-06-02 11:48:59 - [SPOSiteCollectionAdminCSOM] - Processing 'https://redacted.sharepoint.com/sites/PrsentationsGBM' setting 'admin-sp@redacted.qc.ca' IsSiteAdmin 'True'
2024-06-02 11:48:59 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:59 - [AppInfo] - Got access token from memory
2024-06-02 11:48:59 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:59 - [SPOSiteUserCSOM] - Getting user with email 'lperez@redacted.qc.ca' from Site 'https://redacted.sharepoint.com/sites/PrsentationsGBM'
2024-06-02 11:48:59 - [AppInfo] - Getting Access Token for root site https://redacted.sharepoint.com
2024-06-02 11:48:59 - [AppInfo] - Got access token from memory
2024-06-02 11:48:59 - [AppInfo] - Access Token expiration time: 2024-06-02 12:50:03 PM +00:00
2024-06-02 11:48:59 - [SPOSiteUserCSOM] - User with email 'lperez@redacted.qc.ca' no found in Site 'https://redacted.sharepoint.com/sites/PrsentationsGBM'
2024-06-02 11:48:59 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://redacted.sharepoint.com/sites/m-3884'
2024-06-02 11:48:59 - [SPOSiteCollectionAdminCSOM] - Processing 'https://redacted.sharepoint.com/sites/m-3884' setting 'admin-sp@redacted.qc.ca' IsSiteAdmin 'True'
2024-06-02 11:48:59 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-06-02 11:48:59 - [AppInfo] - Got access token from memory
2024-06-02 11:48:59 - [AppInfo] - Access Token expiration time: 2024-06-02 1:07:56 PM +00:00
2024-06-02 11:48:59 - [AppInfo] - Getting Access Token for root site https://redacted.sharepoint.com
2024-06-02 11:48:59 - [AppInfo] - Got access token from memory
2024-06-02 11:48:59 - [AppInfo] - Access Token expiration time: 2024-06-02 12:50:03 PM +00:00
2024-06-02 11:49:00 - [NPLogger] - Error processing Site 'https://redacted.sharepoint.com/sites/m-3884'
2024-06-02 11:49:00 - [NPLogger] - Exception: Les ajouts à ce site web ont été bloqués.
Contactez l'administrateur pour résoudre ce problème.
2024-06-02 11:49:00 - [NPLogger] - Trace:    at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOSiteCollectionAdminCSOM.SetAsync(String siteUrl, String userAdmin, Boolean isSiteAdmin)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOSiteCollectionAdminCSOM.SetAsync(String siteUrl, String userAdmin)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOTenantSiteUrlsWithAccessCSOM.ProcessSiteAsync(SPOTenantSiteUrlsRecord record, String adminUPN)+MoveNext()
2024-06-02 11:49:00 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://redacted.sharepoint.com/sites/Travauxdoptimisationenimageriemdicale'
2024-06-02 11:49:00 - [SPOSiteCollectionAdminCSOM] - Processing 'https://redacted.sharepoint.com/sites/Travauxdoptimisationenimageriemdicale' setting 'admin-sp@redacted.qc.ca' IsSiteAdmin 'True'
2024-06-02 11:49:00 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com

English text of the error here:

2024-06-02 11:49:30 - [NPLogger] - Error processing Site 'https://redacted.sharepoint.com/sites/m-5563'
2024-06-02 11:49:30 - [NPLogger] - Exception: Additions to this Web site have been blocked.
Please contact the administrator to resolve this problem.
2024-06-02 11:49:30 - [NPLogger] - Trace:    at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOSiteCollectionAdminCSOM.SetAsync(String siteUrl, String userAdmin, Boolean isSiteAdmin)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOSiteCollectionAdminCSOM.SetAsync(String siteUrl, String userAdmin)
   at NovaPointLibrary.Commands.SharePoint.Site.SPOTenantSiteUrlsWithAccessCSOM.ProcessSiteAsync(SPOTenantSiteUrlsRecord record, String adminUPN)+MoveNext()

**Comment by Barbarur** [comment]
Can you check that site? It might be set to read-only or blocked access.

**Comment by julesdadrummer** [comment]
The user account I'm using has access to all SharePoint sites and it worked great originally. 

I haven't been able to find anything that would make the site read-only/blocked but I'll continue looking. Is it possible we've hit some sort of Microsoft throttling threshold due to the amount of queries required?

**Comment by Barbarur** [comment]
Throttling threshold shouldn't be, the error would be different.

Most likely it's some setting on the site. Though difficult to say with this generic message. SharePoint also throws one-time issues sometimes.

If you run the solution for that specific site only. Does the issue reproduce?

**Comment by julesdadrummer** [comment]
Hi,

If i only run the solution for 1 site, the issue does not reproduce. I'll do some further testing but the end goal in our case is to go through the ~1500 sharepoint sites and validate which sites a specific user has access to as the mapping of user to site is not well documented and we need to be sure of the access we are granting. 

Thanks!

**Comment by Barbarur** [comment]
Hello @julesdadrummer ,

Did you get any more clarity from your test on the affected site? I got similar error message when using PowerShell for a read-only site, though using the app gives a slightly different error message. Still not sure what could be affecting your site.

It's strange as the actions when processing a site individually are the same as when processing all sites.

The error you are encountering is when adding your account as Site Collection Admin. For the moment, I implemented an option to skipt this action if you are already Site Collection admin (or run the Solution [Add or Remove user as Site Collection Admin](https://github.com/Barbarur/NovaPoint/wiki/Solution-Automation-SetSiteCollectionAdminAuto) in advance). I should help you to avoid the error and the solution will run a bit faster.
